### ISSUE TITLE
Repository

### ISSUE THREAD

**Issue by seniorm0ment** [Musi13/jellyfin-plugin-streamlink#1]
Would be nice to have a Jellyfin repository created for this.

**Comment by PrplHaz4** [comment]
> 
> 
> Would be nice to have a Jellyfin repository created for this.

I'm guessing the reason is that it requires modifications to the jellyfin-web client.

@Musi13 it may be possible to add the functionality similar to ServerWMC or IPTV plugins as a "Channel" instead of under LiveTV...

https://github.com/jellyfin-archive/jellyfin-plugin-iptv

**Comment by azagon63** [comment]
I haven't done anything with the iptv plugin, but that does seem like a better way to do this. In that case there wouldn't be any changes to jellyfin-web since adding/removing streamlink streams would happen in the plugin config.

I'll take a look at refactoring over the next week or so

**Comment by azagon63** [comment]
I pushed release 2.0.0 which switches to the Channel provider; as far as I can tell there aren't any new issues introduced, but if you notice something feel free to open an issue.

I did notice that the still-somewhat-beta Roku app doesn't have support for Channels outside of LiveTV yet, but I think that's a pretty easy change.

Edit: Oh and I still don't plan to request this be added to the larger Jellyfin project. I think its closer to being 'production ready', but I'd still like to have better error reporting and other QoL features before even marking releases as something other than pre-release.
